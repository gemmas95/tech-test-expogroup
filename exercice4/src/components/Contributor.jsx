import React from "react";

// To render contributors change user.name for user.login change the commented loadContributors in logic/contributorsActions
function Contributor({ user }) {
  return <li data-testid="user__login">{user.login}</li>;
}
export default Contributor;
