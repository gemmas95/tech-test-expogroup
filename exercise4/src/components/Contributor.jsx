import React from "react";
import "./contributor.css";

// To render contributors change user.name for user.login change the commented loadContributors in logic/contributorsActions
function Contributor({ user }) {
  console.log("this is user", user);
  return (
    <li className="profile-card">
      <img src={`${user.avatar_url}`} alt={user.id}></img>
      <p
        data-testid="user__login"
        className="text-uppercase text-monospace font-weight-bold"
      >
        {user.login}
      </p>
      <p>Number of contributions: {user.contributions}</p>
      <a href={user.html_url} className="btn btn-dark">
        {" "}
        Link to profile
      </a>
    </li>
  );
}
export default Contributor;
