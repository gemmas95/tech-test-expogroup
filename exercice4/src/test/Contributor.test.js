import React from "react";
import renderer from "react-test-renderer";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import Contributor from "../components/Contributor";

describe("Contributor", () => {
  let container = null;
  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });
  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it("should render a contributor", () => {
    const fakeContributor = { login: "name" };

    act(() => {
      render(<Contributor user={fakeContributor} />, container);
    });

    expect(
      container.querySelector('[data-testid="user__login"]').textContent
    ).toBe(fakeContributor.login);
  });
  it("should match Contributor component", () => {
    const fakeContributor = { login: "name", id: 12345 };
    const contributorTree = renderer.create(
      <Contributor user={fakeContributor} key={fakeContributor.id} />
    );

    expect(contributorTree).toMatchSnapshot();
  });
});
