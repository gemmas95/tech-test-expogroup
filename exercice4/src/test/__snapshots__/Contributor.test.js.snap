// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contributor should match Contributor component 1`] = `
<li
  data-testid="user__login"
>
  name
</li>
`;
